src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -399,6 +409,58 @@ bool CanApplyChange(DocumentId documentId)
             }
         }
 
+        private void NotifyRefactorChanges(IEnumerable<DocumentId> changedDocs, CodeAnalysis.Solution oldSolution)
+        {
+            // Without any services to notify there's no need to dig through the documents
+            if (!_refactorNotifyServices.Any())
+            {
+                return;
+            }
+
+            foreach (var changedDocId in changedDocs)
+            {
+                var newDocument = CurrentSolution.GetDocument(changedDocId);
+
+                // Documents without syntax tree won't have the annotations attached
+                if (!newDocument.SupportsSyntaxTree)
```

Comment:
Want to filter out any documents also not supporting SemanticModels, since we won't be able to handle those either?